src/dbnode/persist/fs/msgpack/decoder.go

```diff
@@ -404,9 +417,14 @@ func (dec *Decoder) decodeIndexEntry(bytesPool pool.BytesPool) schema.IndexEntry
 	// final field on index entries
 	dec.skip(numFieldsToSkip)
 
+	actualChecksum := dec.checksumAndUnwrapStreamWithDigest()
```

Comment:
Should we add a comment here about always leaving the index entry checksum last? 

Btw, Im not familiar with msgpack encoding but IIRC proto encodings are always backwards compatible but fields are always appended and never added in the middle. 

Also, what was the rationale for using msgpack in the first place (instead of proto)? Was it performance related or did we had to do manual parsing for some reason and it was easier to do that using msgpack?

Reply:
Re: index entry checksum last -- Yeah, I kinda allude to that in the comment on lines 416-417, but I think I can make the call out a little stronger. Will add a better comment here

Re: backwards compatibility -- That's true about protos. With msgpack, we have a little more control, well, I guess we have total control, over how we encode and decode our objects. With proto, you provide an object and protobuf turns it into a set of bytes corresponding to that object. With msgpack, we're responsible for how we lay out that object as bytes, msgpack just helps us with the actual encoding of primitives as bytes. Given that, we can play games like set a persistent footer (i.e. checksum) because we own how we inflate the object from the bytes. We can use the version of the code combined with the version of the file we're processing to know how to skip bytes we might not be interested in. That's basically what we do here to allow for forward and backward compatibility.

Re: msgpack over proto -- That's more of an @robskillington question. We briefly talked about it and I think the answer was simplicity, though one of the drawbacks he mentioned is that you end up re-implementing a lot of what protos already gives you (the discussion in the previous paragraph is a good example of that).

Reply:
Added comment on L422